<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PretrainedModelPage_Detectron2</class>
 <widget class="QWizardPage" name="PretrainedModelPage_Detectron2">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>550</height>
   </rect>
  </property>
  <property name="title">
   <string>Pre-trained Model Selection</string>
  </property>
  <property name="subTitle">
   <string>Choose how to initialize your Detectron2 model weights.</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <item>
    <widget class="QGroupBox" name="modeGroupBox">
     <property name="title">
      <string>Would you like to use a pre-trained model?</string>
     </property>
     <layout class="QVBoxLayout" name="modeLayout">
      <item>
       <widget class="QRadioButton" name="downloadRadio">
        <property name="text">
         <string>Download pre-trained model (Recommended for new users)</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="downloadDescription">
        <property name="styleSheet">
         <string notr="true">color: #666; margin-left: 20px;</string>
        </property>
        <property name="text">
         <string>Start with a model trained on COCO dataset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="scratchRadio">
        <property name="text">
         <string>Start from scratch</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="scratchDescription">
        <property name="styleSheet">
         <string notr="true">color: #666; margin-left: 20px;</string>
        </property>
        <property name="text">
         <string>Train your own model from random weights</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="existingRadio">
        <property name="text">
         <string>Use existing model file</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="existingDescription">
        <property name="styleSheet">
         <string notr="true">color: #666; margin-left: 20px;</string>
        </property>
        <property name="text">
         <string>Select a .pt/.pth file from disk</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="pretrainedModelsWidget" native="true">
     <layout class="QVBoxLayout" name="pretrainedModelsLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="modelsLabel">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Available pre-trained models:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="cocoFrame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <layout class="QVBoxLayout" name="cocoLayout">
         <item>
          <widget class="QRadioButton" name="cocoRadio">
           <property name="font">
            <font>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>COCO Instance Segmentation (Recommended)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="cocoInfo">
           <property name="styleSheet">
            <string notr="true">color: #666; margin-left: 20px;</string>
           </property>
           <property name="text">
            <string>80 classes (person, car, dog, etc.)
Size: ~178 MB
mAP: 37.2 on COCO val</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="lvisFrame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <layout class="QVBoxLayout" name="lvisLayout">
         <item>
          <widget class="QRadioButton" name="lvisRadio">
           <property name="text">
            <string>LVIS Instance Segmentation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lvisInfo">
           <property name="styleSheet">
            <string notr="true">color: #666; margin-left: 20px;</string>
           </property>
           <property name="text">
            <string>1203 classes (more detailed categories)
Size: ~182 MB
mAP: 25.6 on LVIS val</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="cityscapesFrame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <layout class="QVBoxLayout" name="cityscapesLayout">
         <item>
          <widget class="QRadioButton" name="cityscapesRadio">
           <property name="text">
            <string>Cityscapes (Urban scenes)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="cityscapesInfo">
           <property name="styleSheet">
            <string notr="true">color: #666; margin-left: 20px;</string>
           </property>
           <property name="text">
            <string>8 classes (car, pedestrian, etc.)
Size: ~175 MB
Optimized for street scenes</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="existingModelWidget" native="true">
     <layout class="QHBoxLayout" name="existingModelLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLineEdit" name="existingModelEdit">
        <property name="placeholderText">
         <string>Select model file...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="browseExistingButton">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
